src/content/core/build-service.mdx

```diff
@@ -130,6 +130,18 @@ Instead, the image `david/api:okteto` is re-tagged to `okteto/api:09f8e` to reme
   />
 </p>
 
+### Prerequisite: Git environment required
+
+Smart Builds needs access to a git repository to compute the content hash correctly. It uses more than just the branch and commit; it also inspects diffs, untracked files, and folder content to determine whether a build can be skipped.
+
+:::caution
+Do not exclude the `.git` directory from your `.oktetoignore` if you want Smart Builds to work. If `.git` is excluded, Smart Builds cannot access git metadata and will fall back to a full build.
```

Comment:
This is only for the case `if your manifest runs Okteto CLI commands that trigger image builds (for example `deploy: commands:` that run `okteto deploy -f compose.yml`)`. I would unify this note with this sentence:
https://github.com/okteto/docs/pull/1085/files#diff-a679e4f67f537215eb823ca96ec5e6c837de3b4321ae93ac6c49c7e92b1ca230R143